src/linker/Linker.Steps/LinkAttributesParser.cs

```diff
@@ -33,35 +33,44 @@ public void Parse (AttributeInfo xmlInfo)
 			ProcessXml (stripLinkAttributes, _context.IgnoreLinkAttributes);
 		}
 
-		IEnumerable<CustomAttribute> ProcessAttributes (XPathNavigator nav, ICustomAttributeProvider provider)
+		CustomAttribute[] ProcessAttributes (XPathNavigator nav, ICustomAttributeProvider provider)
 		{
 			XPathNodeIterator iterator = nav.SelectChildren ("attribute", string.Empty);
-			var attributes = new List<CustomAttribute> ();
+			var builder = new ArrayBuilder<CustomAttribute> ();
 			while (iterator.MoveNext ()) {
 				if (!ShouldProcessElement (iterator.Current))
 					continue;
 
+				TypeDefinition attributeType;
 				string internalAttribute = GetAttribute (iterator.Current, "internal");
 				if (!string.IsNullOrEmpty (internalAttribute)) {
-					ProcessInternalAttribute (provider, internalAttribute);
-					continue;
-				}
+					var removeAttributeType = typeof (RemoveAttributeInstancesAttribute);
+					var removeAttributeTypeAssembly = AssemblyDefinition.ReadAssembly (removeAttributeType.Assembly.Location);
+					var tr = removeAttributeTypeAssembly.MainModule.ImportReference (removeAttributeType);
```

Comment:
This loads linker's assembly into the closure of the app being trimmed. This might cause trouble - for example trying to run 6.0 linker on a 5.0 app might be problematic.
I think it would be easier/cleaner if we simply defined the attribute anew in one of the already loaded modules (CoreLib?).